ext/nnstreamer/tensor_filter/tensor_filter_flare.cc

```diff
@@ -0,0 +1,386 @@
+/**
+ * GStreamer Tensor_Filter, flare Module
+ * Copyright (C) 2025 Hyunil Park <hyunil46.park@samsung.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * @file    tensor_filter_flare.cc
+ * @date    2 July 2025
+ * @brief   flare module for tensor_filter gstreamer plugin
+ * @see     http://github.com/nnstreamer/nnstreamer
+ * @author  Hyunil Park <hyunil46.park@samsung.com>
+ * @bug     No known bugs except for NYI items
+ */
+
+#include <SFlareApi.h>
+#include <assert.h>
+#include <atomic>
+#include <condition_variable>
+#include <errno.h>
+#include <functional>
+#include <iostream>
+#include <memory>
+#include <nnstreamer_cppplugin_api_filter.hh>
+#include <nnstreamer_log.h>
+#include <nnstreamer_plugin_api_util.h>
+#include <nnstreamer_util.h>
+#include <string>
+#include <vector>
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+void init_filter_flare (void) __attribute__ ((constructor));
+void fini_filter_flare (void) __attribute__ ((destructor));
+#ifdef __cplusplus
+}
+#endif /* __cplusplus */
+
+namespace nnstreamer
+{
+namespace tensorfilter_flare
+{
+/**
+ * @brief Class for TensorFilterFlare subplugin
+ */
+class TensorFilterFlare : public tensor_filter_subplugin
+{
+  public:
+  TensorFilterFlare ();
+  ~TensorFilterFlare ();
+
+  /* mandatory methods */
+  tensor_filter_subplugin &getEmptyInstance ();
+  void configure_instance (const GstTensorFilterProperties *prop);
+  void invoke (const GstTensorMemory *input, GstTensorMemory *output);
+  void invoke_dynamic (GstTensorFilterProperties *prop,
+      const GstTensorMemory *input, GstTensorMemory *output);
+  void getFrameworkInfo (GstTensorFilterFrameworkInfo &info);
+  int getModelInfo (model_info_ops ops, GstTensorsInfo &in_info, GstTensorsInfo &out_info);
+  int eventHandler (event_ops ops, GstTensorFilterFrameworkEventData &data);
+
+  /* static methods */
+  static void init_filter_flare ();
+  static void fini_filter_flare ();
+
+  private:
+  static TensorFilterFlare *registered;
+  static const char *name;
+
+  SFlareApi::SFlareContext *context;
+  SFlareApi::SFlareConfig config;
```

Comment:
Not ensured to be initialized.
```
Config initialization:
SFlareApi::SFlareConfig config; is not zero-initialized; only some fields are set. Initialize with {} in the constructor to avoid uninitialized values.
```